addons/account/models/account_move.py

```diff
@@ -3230,6 +3223,55 @@ def _get_new_hash(self, secure_seq_number):
         #build and return the hash
         return self._compute_hash(prev_move.inalterable_hash if prev_move else u'')
 
+    def _hash_moves(self, force=False):
+        self.flush_recordset()  # Ensure that the name is correctly computed before it is used to generate the hash
+        for move in self._get_moves_to_hash(force=force):
+            new_number = move.journal_id.secure_sequence_id.next_by_id()
+            super(AccountMove, move).write({
+                'secure_sequence_number': new_number,
+                'inalterable_hash': move._get_new_hash(new_number),
+            })
+
+    def _get_moves_to_hash(self, force=False):
+        """
+        :return: the moves that need to be hashed taking into account all moves after the last
+                 hashed moved in the chain.
+        """
+        self = self.sorted(lambda m: (m.ref or '', m.id))[-1:]
+        if (
+            not self
+            or
+            not (
+                (self.move_type in self.get_sale_types(include_receipts=True) and self.is_move_sent)
+                or
+                (self.move_type in self.get_purchase_types(include_receipts=True) and self.state == 'posted')
+                or
+                self.move_type == 'entry'
+                or
+                force
+            )
+        ):
+            return self.env['account.move']
+        domain = [
+            ('restrict_mode_hash_table', '=', True),
+            ('state', '=', 'posted'),
+            ('inalterable_hash', '=', False),
+            ('journal_id', '=', self.journal_id.id),
+            ('date', '>', self.company_id.fiscalyear_lock_date) if self.company_id.fiscalyear_lock_date else (1, '=', 1),
```

Comment:
Use `_get_user_fiscal_lock_date` for compatibility with the company branches

Reply:
Cool, I can even get rid of the trick 1=1